Fix phpunit deprecations

eerison opened the pull request:
<!-- THE PR TEMPLATE IS NOT AN OPTION. DO NOT DELETE IT, MAKE SURE YOU READ AND EDIT IT! -->
## Fix phpunit deprecations


- it is blocked by https://github.com/SymfonyTest/SymfonyDependencyInjectionTest/pull/165
- `assertContainsOnlyInt` was added in phpunit 11.5.0: https://github.com/sebastianbergmann/phpunit/commit/a726e0396e71cc77bc0b459f93481c29e726dbd8
<!-- Describe your Pull Request content here -->
- phpunit 11 supports `=>8.2`

<!--
    Show us you choose the right branch.
    Different branches are used for different things :
    - 4.x is for everything backwards compatible, like patches, features and deprecation notices
    - 5.x is for deprecation removals and other changes that cannot be done without a BC-break
    More details here: https://github.com/sonata-project/SonataPageBundle/blob/4.x/CONTRIBUTING.md#base-branch
-->

<!--
    Specify which issues will be fixed/closed.
    Remove it if this is not related.
-->


## Changelog

<!-- MANDATORY
    Fill the changelog part inside the code block.
    Follow this schema: https://keepachangelog.com/
    This will end up on https://github.com/sonata-project/SonataPageBundle/releases,
    please keep it short and clear and to the point
-->

<!--
    If you are updating something that doesn't require
    a release, you can delete the whole "Changelog" section.
    (eg. update to docs, tests)
-->

<!-- REMOVE EMPTY SECTIONS
    - Don't repeat the verb in the messages (don't use "Added", "Changed", etc).
    - Don't use a full stop at the end of the messages.
-->
```markdown
### Removed
- Removed support for PHP 8.1
```

<!--
    If this is a work in progress, uncomment the "To do" section.
    You can add as many tasks as you want.
    If some are not relevant, just remove them.
-->

eerison committed:
Fix phpunit deprecations

it is fixing deprecations related with phpunit 13

Removed deprecated supress

Remove php 8.1 support

Remove runtime test

update dependencies

run php fix

eerison commented on .github/workflows/test.yaml:
```diff
@@ -30,7 +30,6 @@ jobs:
         strategy:
             matrix:
                 php-version:
-                    - '8.1'
```
I know I can not change this file, it is just for test :)

dmaicher commented on composer.json:
```diff
@@ -21,7 +21,7 @@
     "homepage": "https://docs.sonata-project.org/projects/SonataPageBundle",
     "require": {
-        "php": "^8.1",
+        "php": "^8.2",
```
we are planning to drop support for 8.1 anyway when adding Symfony 8 support. So this change is fine :+1:

eerison commented:
there is a issue introduced in symfony/panther, I need to figure out how to handle it

https://github.com/symfony/panther/issues/169
https://github.com/symfony/panther/issues/436

 ```
 Symfony\Component\Panther\Exception\RuntimeException: The port 9080 is already in use.
 ```

dmaicher commented:
You can ignore the psalm fails as we dropped it anyway in #1835 

Do you maybe want to pull in those changes? could cherry pick those commits

dmaicher commented:
phpstan now runs on php 8.5 hence some new errors

eerison committed:
Remove Psalm

eerison commented:
> phpstan now runs on php 8.5 hence some new errors

let's add php 8.5 and symfony 8 in other PR

eerison commented on composer.json:
```diff
@@ -80,7 +80,8 @@
         "doctrine/dbal": "<3.4",
-        "doctrine/orm": "<2.14"
+        "doctrine/orm": "<2.14",
+        "symfony/dom-crawler": ">7.4"
```
it can be reverted after this issue be solved

https://github.com/symfony/panther/issues/685

eerison committed:
replace class_exists to interface_exists

eerison committed:
Remove enable_native_lazy_objects option in tests

Since doctrine/doctrine-bundle 3.1: The "enable_native_lazy_objects"
option is deprecated and will be removed in DoctrineBundle 4.0, as
native lazy objects are now always enabled.

dmaicher commented on tests/Runtime/SonataPagePathRuntimeTest.php:
```diff
@@ -1,53 +0,0 @@
- * NEXT_MAJOR: Remove this class.
- */
-#[IgnoreDeprecations]
-final class SonataPagePathRuntimeTest extends TestCase
```
hm do we need to delete this test?

eerison replied:
Yep 😄 , it was returning a strange error, and as it was marked to remove on next release then I just removed it.

edit: I do not remember the error anymore, I will revert to see it again :)

eerison replied:
Ok, somehow it is working again 😄

dmaicher commented on composer.json:
```diff
@@ -60,14 +60,14 @@
-        "phpunit/phpunit": "^10.5.54 || ^11.5.38 || ^12.3.10",
+        "phpunit/phpunit": "^11.5.38 || ^12.3.10",
         "psalm/plugin-phpunit": "^0.19",
         "psalm/plugin-symfony": "^5.0",
```
shall we remove psalm dependencies and all `@psalm-supress...` annotations then in a follow-up PR?

dmaicher commented:
Thank you @eerison
